packages/back-end/src/services/features.ts

```diff
@@ -54,11 +78,21 @@ export async function getFeatureDefinitions(
       rules:
         settings.rules
           ?.filter((r) => r.enabled)
-          ?.map((r) => {
+          ?.map(async (r) => {
             const rule: FeatureDefinitionRule = {};
             if (r.condition && r.condition !== "{}") {
               try {
                 rule.condition = JSON.parse(r.condition);
+                for (const attribute in rule.condition) {
+                  for (const condition in rule.condition[attribute]) {
+                    if (condition === ("$inGroup" || "$notInGroup")) {
+                      const updatedCondition = await formatCondition(
+                        rule.condition[attribute]
+                      );
```

Comment:
How many conditions are typically run in an operation like this? It would be more efficient to make a bulk request to MongoDB up front rather than for each item. If there's 100 conditions, then we would be making 99 more round trips to the database than necessary.

As a new person who's never seen this code, it's not exactly clear what you mean by `formatCondition`.

Reply:
Great suggestion and this is exactly the route we took. Resolving this comment.